File: spring-boot-rest/src/main/scala/com/stulsoft/poc/rest/Application.scala
/*
 * Copyright (c) 2018. Yuriy Stul
 */

package com.stulsoft.poc.rest

import org.slf4j.LoggerFactory
import org.springframework.boot.SpringApplication

/**
  * @author Yuriy Stul
  */
object Application {
  private val logger = LoggerFactory.getLogger(Application.getClass)

  def main(args: Array[String]): Unit = {
    logger.info("==>main")
    val context = SpringApplication.run(classOf[ApplicationConfiguration], args: _*)
    logger.info("<==main")
  }
}

File: pgradle/gradle-spring/app/src/main/java/com/stulsoft/poc/gradle/spring/Application.java
/*
 * Copyright (c) 2021. StulSoft
 */

package com.stulsoft.poc.gradle.spring;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.core.env.AbstractEnvironment;

/**
 * @author Yuriy Stul
 */
@SpringBootApplication
public class Application implements CommandLineRunner {
    private static final Logger logger = LoggerFactory.getLogger(Application.class);

    private final String abc;

    public Application(@Autowired final String abc) {
        this.abc = abc;
    }

    public static void main(String[] args) {
        logger.info("==>main");
        System.setProperty(AbstractEnvironment.DEFAULT_PROFILES_PROPERTY_NAME, "dev");
        SpringApplication.run(Application.class, args);
    }

    @Override
    public void run(String... args) {
        logger.info("==>run");
        logger.info("abc={}", abc);
    }
}
